scripts/shared.js

```diff
@@ -605,14 +605,47 @@ async function queryMatchingArticles(
   return articles;
 }
 
+/**
+ * Queries the site's index and returns articles whose specified metadata value
+ * matches a given filter value.
+ * @param {string} metadataName Name of the article metadata to match.
+ * @param {string} value Value to match with.
+ * @param {number} pageNum Current Page Number.
+ * @param {function} [compareFn] Used to compare the two values. The first parameter will be
+ *  the record's metadata value, and the second will be a lower-case version of the match
+ *  value. Default function will convert the record's metadata value to lower case and strictly
+ *  compare it with the match value.
+ * @returns {Promise<QueryIndexRecord>} Article's record information.
+ */
+async function queryMatchingArticlesByCategory(
```

Comment:
Rather than copy/pasting a lot of what we've already got, would it be possible to just add a new parameter for the offset to `queryMatchingArticles()`?

Reply:
@mfrisbey, the existing implementation sets a chunk limit value of 3. This prevents all the articles from being fetched and causes the Next button to be disabled even when we have more articles belonging to the specific category. Removing the chunk limit at global level might affect other functionality. as such I decided to go ahead with a separate method